File: c/5-Arrays/practice/ap-4.c
#include <stdio.h>
int main()
{
    int a;
    if(scanf("%d",&a)!=1||a<=0)
    {
        printf("Invalid Input");
        return 0;
    }
    
    int arr[a][a];
    for(int i =0;i<a;i++)
    {
        for(int j =0;j<a;j++)
        {
            scanf("%d ",&arr[i][j]);
        }printf("\n");
    }
    
    int sum =0;
    int product =1;
    for(int i =0;i<a;i++)
    {
        for(int j =i;j==i;j++)
        {
           sum+=arr[i][j];
           product*=arr[i][j];
        }
    }
    printf("Sum:%d\n",sum);
    printf("Product:%d",product);
    
}

File: java/4-Arrays/practice/ap-4.java
import java.util.Scanner;
public class P{
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);
        int a =sc.nextInt();
        int b =sc.nextInt();
        if(a<0||b<0)
        {
            System.out.println("Invalid input");
            return;
        }
        int arr[][] = new int[a][b];
        for(int i =0 ;i<a;i++)
        {
            for(int j=0;j<b;j++)
            {
                if(sc.hasNextInt()){
                arr[i][j] =sc.nextInt();
                }
                else{
                    System.out.println("Invalid input");
                    return;
                }
            }
            System.out.println();
        }
          for(int i =0 ;i<b;i++)
        {
            int sum =0;
            for(int j=0;j<a;j++)
            {
               sum+=arr[j][i];
            }
            System.out.println(sum);
        }
        
        
    }
}